From 2adb32b012c33c385e027ce6ea0c61e328117b74 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 23 Sep 2014 14:56:09 -0400
Subject: [drm] bochs: Remove unecessary NULL check in gem_free

Message-id: <1411484172-28849-7-git-send-email-kraxel@redhat.com>
Patchwork-id: 96485
O-Subject: [RHEL-7.1 kernel PATCH 6/9] drm/bochs: Remove unecessary NULL check in gem_free
Bugzilla: 1043379
RH-Acked-by: Dave Airlie <airlied@redhat.com>

From: Daniel Vetter <daniel.vetter@ffwll.ch>

The ->gem_free_object never gets called with a NULL pointer, the check
is redundant. Also checking after the upcast allows compilers to elide
it anyway.

Noticed while chasing coverity reports, somehow this one here was not
flagged.

Reviewed-by: David Herrmann <dh.herrmann@gmail.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
(cherry picked from commit e0c6a73fb191daad6d4ea808a89c6e22ac1b2733)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/bochs/bochs_mm.c b/drivers/gpu/drm/bochs/bochs_mm.c
index ec48746..c1c6337 100644
--- a/drivers/gpu/drm/bochs/bochs_mm.c
+++ b/drivers/gpu/drm/bochs/bochs_mm.c
@@ -439,8 +439,6 @@ void bochs_gem_free_object(struct drm_gem_object *obj)
 {
  struct bochs_bo *bochs_bo = gem_to_bochs_bo(obj);
 
- if (!bochs_bo)
-  return;
  bochs_bo_unref(&bochs_bo);
 }
 
-- 
1.7.1